assets/src/stories-editor/blocks/amp-story-text/edit.js

```diff
@@ -123,7 +123,9 @@ class TextBlockEdit extends Component {
 					// Ensure line breaks are normalised to HTML.
 					value={ content }
 					onChange={ ( nextContent ) => setAttributes( { content: nextContent } ) }
+					// The 2 following lines are necessary for pasting to work.
 					onReplace={ this.onReplace }
+					onSplit={ () => {} }
```

Comment:
Looks like for replacing to work properly `onSplit` needs to be defined, too, we're not actually using it, so it should work with an empty function.